PowerValidatedSolutions.psm1

```diff
@@ -48145,6 +48145,383 @@ Function Test-WMSubnetInput {
+				Remove-Item $cfgFilePath
+				Remove-Item $orginalKeyFilePath
+				break;
+			}
+        } else {
+			if($createRandomCounter -gt 6) {
+				Write-Error "Error: Unable to create temporary files in directory $certDir." -ErrorAction Stop
+				Exit
+			}
+        }
+    }
+
+    $keyFilePath = Join-Path -path $certDir -ChildPath "$commonName.key"
+    $csrFilePath = Join-Path -path $certDir -ChildPath "$commonName.csr"
+    if (Test-Path -Path $keyFilePath) {
+        Write-Error "Error: $keyFilePath file already exists.  Please remove/rename the current file or find a new certificate directory" -ErrorAction Stop
+    }
+    if (Test-Path -Path $csrFilePath) {
+        Write-Error "Error: $csrFilePath file already exists.  Please remove/rename the current file or find a new certificate directory" -ErrorAction Stop
+    }
+    if ($expireDays -lt 2) {
+        Write-Error "Error: certificate expiration day($expireDays) is less than 1 day.  Please provide a value more than 1 day."
+    }
+
+    # Create temporary cfg file
+    $cfgString = '[ req ]
+    days = ' +$expireDays+ '
+    default_md = sha512
+    default_bits = ' +$keySize+ '
+    distinguished_name = req_distinguished_name
+    prompt = no
+    req_extensions = v3_req
+    
+    [ v3_req ]
+    basicConstraints = CA:FALSE
+    keyUsage = digitalSignature, keyEncipherment, dataEncipherment, nonRepudiation
+    extendedKeyUsage = serverAuth, clientAuth
+    subjectAltName = @alternate_names
+
+    [ req_distinguished_name ]
+    commonName = ' +$commonName+ '
+    countryName = ' +$country+ '
+    stateOrProvinceName = ' +$state+ '
+    localityName = ' +$locality+ '
+    0.organizationName = ' +$organization+ '
+    organizationalUnitName = ' +$organizationUnit+ '
+    
+    [ alternate_names ]
+    '
+    # Parsing SAN string
+    $sanArrary = $subjectAlternativeNames -split "\s*,\s*"
+    $sanCount = 1
+
+    foreach($san in $sanArrary) {
+        $cfgString = $cfgString + "DNS." + $sanCount + " = " + $san + "`n"
+        $sanCount++
+    }
+
+    # Generate RSA private key and CSR file
+    $cfgString | Out-File $cfgFilePath
+    openssl req -new -sha256 -nodes -out $csrFilePath -keyout $orginalKeyFilePath -config $cfgFilePath >> $null 2>$null
+    if ($opensslVersion -match "OpenSSL 3.") {
+        openssl rsa -in $orginalKeyFilePath -out $keyFilePath -traditional >> $null 2>$null
+    } else {
+        openssl rsa -in $orginalKeyFilePath -out $keyFilePath >> $null 2>$null
+    }
+
+    # Clean up
+    remove-item $cfgFilePath
+    remove-item $orginalKeyFilePath
+    
+}
+Export-ModuleMember -Function Invoke-GeneratePrivateKeyAndCsrFiles
+
+Function Invoke-RequestSignedCertificateByMsca {
+    <#
+        .SYNOPSIS
+        Request CSR to be signed by a MSCA Certificate Authority.
+
+        .DESCRIPTION
+        The Invoke-RequestSignedCertificateByMsca cmdlet connects to MSCA web enrollment to sign a CSR.  
```

Comment:
```suggestion
        The Invoke-RequestSignedCertificateByMsca cmdlet connects to Microsoft Certificate Authority web enrollment service to sign a certificate signing request (CSR).  
```

Reply:
Addressed in commit 53039acbaaffafaf219274cac1a3684735130f53